src/ex1/script.js

```diff
@@ -0,0 +1,141 @@
+let taskId = 0;
+let tasksCnt = 0;
+let isAddTaskClickable = false;
+const addTaskInput = document.querySelector(".add-task-input");
+const addTaskBtn = document.querySelector(".add-task-btn");
+
+// Add task will be clickable only for a non empty task
+addTaskInput.addEventListener("input", () => {
+  if (addTaskInput.value && !isAddTaskClickable) {
+    addTaskBtn.classList.add("valid-task-btn");
+    isAddTaskClickable = true;
+  } else if (!addTaskInput.value) {
+    addTaskBtn.classList.remove("valid-task-btn");
+    isAddTaskClickable = false;
+  }
+});
+
+// Add task
+addTaskBtn.addEventListener("click", (event) => onAddTask(event));
+document.addEventListener("keypress", (event) => {
+  if (event.key === "Enter") {
+    const inputElem = document.querySelector(".add-task-input");
+    if (inputElem === document.activeElement) {
```

Comment:
nice! you can get the element from `event.target` as well :)

Reply:
Not sure how? Just replacing inputElem in the second if with event.target won't get the following desired behaviour: when 'enter' is clicked and the input element isn't focused -> don't add a new task. Can you please elaborate on your suggested solution?

Reply:
`event.target` will hold the element that the event is related to.
so, if the user had focus on the input and pressed enter, `event.target` will hold the input element.
not necessary to replace your code, just thought it's a good thing to know